daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -996,4 +1018,90 @@ private[lf] final class Compiler(
           }
         }
     }
+
+  private[this] def translateDisclosedContract(
```

Comment:
FIXME: what about disclosure keys - do we need to cache these in a similar manner?